src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SDSEarlyReleaseDefaultingRulesService.kt

```diff
@@ -84,27 +84,44 @@ class SDSEarlyReleaseDefaultingRulesService(
     return if (dates[ReleaseDateType.ARD] != null) ReleaseDateType.ARD to dates[ReleaseDateType.ARD] else ReleaseDateType.CRD to dates[ReleaseDateType.CRD]
   }
 
-  private fun handleCRDorARDandPRRD(
+  private fun handleCRDorARDAndPRRD(
     dates: MutableMap<ReleaseDateType, LocalDate>,
+    otherDates: MutableMap<ReleaseDateType, LocalDate>,
   ) {
-    if ((dates.containsKey(ReleaseDateType.CRD).or(dates.containsKey(ReleaseDateType.ARD)))
-        .and(dates.containsKey(ReleaseDateType.PRRD))
-    ) {
+    if (dates.containsKey(ReleaseDateType.CRD) || dates.containsKey(ReleaseDateType.ARD)) {
       val controllingDate = getControllingDate(dates)
 
-      if (controllingDate.second?.isAfter(dates[ReleaseDateType.PRRD]) == true) {
-        dates.remove(ReleaseDateType.PRRD)
-      } else {
-        dates.remove(controllingDate.first)
+      // override HDCED to the PRRD if the PRRD is later than the HDCED
+      dates[ReleaseDateType.HDCED4PLUS]?.let { hdcedDate ->
+        val prrdDate = dates[ReleaseDateType.PRRD] ?: otherDates[ReleaseDateType.PRRD]
+        prrdDate?.takeIf { it.isAfter(hdcedDate) }?.let {
+          setHDCEDDates(it, dates)
+        }
+      }
+
+      if (dates.containsKey(ReleaseDateType.PRRD)) {
+        if (controllingDate.second?.isAfter(dates[ReleaseDateType.PRRD]) == true) {
+          dates.remove(ReleaseDateType.PRRD)
+        } else {
+          dates.remove(controllingDate.first)
 
-        // PRRD is later than any other release date, therefore can not have a HDCED
-        // See BookingExtractionService#extractMultiple
-        dates.remove(ReleaseDateType.HDCED4PLUS)
-        dates.remove(ReleaseDateType.HDCED)
+          // PRRD is later than any other release date, therefore can not have a HDCED
+          // See BookingExtractionService#extractMultiple
+          removeHDCEDDates(dates)
+        }
       }
     }
   }
 
+  private fun setHDCEDDates(date: LocalDate, dates: MutableMap<ReleaseDateType, LocalDate>) {
+    dates[ReleaseDateType.HDCED4PLUS] = date
+    dates[ReleaseDateType.HDCED] = date
```

Comment:
Only thing I would say is this might cause a couple of problems until HDCED4PLUS has fully taken over from HDCED, as it is possible that there isn't a HDCED date... BUT in the real world it doesn't actually matter as only HDCED4PLUS is returned